### DISCUSSION TITLE
Licensing in files.

### DISCUSSION THREAD

**Discussion by Battojutsu** [LibreSprite/LibreSprite#487] [General]
The file menu.cpp has a license saying "This file is released under the terms of the MIT license." 

I wouldn't want any of my contributions to be used in closed source proprietary software. 

I'll relicense the files I work on as GPLv2 which I believe I can from this post. https://opensource.stackexchange.com/questions/5832/relicensing-an-mit-licensed-project-under-the-gpl-that-has-non-code-contribution

Maybe it's worth a discussion about relicensing the whole project including the headers into GPLv2

Thoughts?

**Comment by amirallami-code** [comment]
To relicense the entire project to GPLv2, you need consent from all contributors. You can change only your own code to GPLv2, but its impact will be limited to those parts. It's best to discuss this with the project's community.

**Comment by Battojutsu** [comment]
I was under the impression that the MIT License allows re-licensing to other licenses. Which is how the Aseprite creator was able to re-license it to a proprietary license. 

So by that logic I think relicensing everything under GPLV3 is alright since everything was under the MIT License. I heard they are compatible. According to https://softwareengineering.stackexchange.com/questions/204410/how-are-gpl-compatible-licenses-like-mit-usable-in-gpl-programs-without-being-su

"Many of the most common free software licenses, such as the original MIT/X license, ... are "GPL-compatible". That is, their code can be combined with a program under the GPL without conflict (the new combination would have the GPL applied to the whole"
